coderd/provisionerdaemons.go

```diff
@@ -0,0 +1,79 @@
+package coderd
+
+import (
+	"net/http"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/db2sdk"
+	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/coderd/httpmw"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// @Summary Get provisioner daemons
+// @ID get-provisioner-daemons
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param organization path string true "Organization ID" format(uuid)
+// @Param tags query object false "Provisioner tags to filter by (JSON of the form {'tag1':'value1','tag2':'value2'})"
+// @Success 200 {array} codersdk.ProvisionerDaemonWithStatus
+// @Router /organizations/{organization}/provisionerdaemons [get]
+func (api *API) provisionerDaemons(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx      = r.Context()
+		org      = httpmw.OrganizationParam(r)
+		tagParam = r.URL.Query().Get("tags")
+		tags     = database.StringMap{}
+		err      = tags.Scan([]byte(tagParam))
+	)
+
+	if tagParam != "" && err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Invalid tags query parameter",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	daemons, err := api.Database.GetProvisionerDaemonsWithStatusByOrganization(
+		ctx,
+		database.GetProvisionerDaemonsWithStatusByOrganizationParams{
+			OrganizationID:  org.ID,
+			StaleIntervalMS: provisionerdserver.StaleInterval.Milliseconds(),
```

Comment:
future suggestion (non-blocking): we could potentially pass this as a query param to allow users to view older provisioner daemons

Reply:
This only affects the status displayed (i.e. offline or not), it won't filter out old daemons although you could filter on status = offline if you only want those.